MekHQ/src/mekhq/campaign/market/contractMarket/AbstractContractMarket.java

```diff
@@ -0,0 +1,312 @@
+            contract.setCommandRights(ContractCommandRights.LIAISON);
+        } else {
+            contract.setCommandRights(ContractCommandRights.INDEPENDENT);
+        }
+    }
+
+    protected void rollSalvageClause(AtBContract contract, int mod, int contractMaxSalvagePercentage) {
+        contract.setSalvageExchange(false);
+        int roll = Math.min(Compute.d6(2) + mod, 13);
+        if (roll < 2) {
+            contract.setSalvagePct(0);
+        } else if (roll < 4) {
+            contract.setSalvageExchange(true);
+            int r;
+            do {
+                r = Compute.d6(2);
+            } while (r < 4);
+            contract.setSalvagePct(Math.min((r - 3) * 10, contractMaxSalvagePercentage));
+        } else {
+            contract.setSalvagePct(Math.min((roll - 3) * 10, contractMaxSalvagePercentage));
+        }
+    }
+
+    protected void rollSupportClause(AtBContract contract, int mod) {
+        int roll = Compute.d6(2) + mod;
+        contract.setStraightSupport(0);
+        contract.setBattleLossComp(0);
+        if (roll < 3) {
+            contract.setStraightSupport(0);
+        } else if (roll < 8) {
+            contract.setStraightSupport((roll - 2) * 20);
+        } else if (roll == 8) {
+            contract.setBattleLossComp(10);
+        } else {
+            contract.setBattleLossComp(Math.min((roll - 8) * 20, 100));
+        }
+    }
+
+    protected void rollTransportClause(AtBContract contract, int mod) {
+        int roll = Compute.d6(2) + mod;
+        if (roll < 2) {
+            contract.setTransportComp(0);
+        } else if (roll < 6) {
+            contract.setTransportComp((20 + (roll - 2) * 5));
+        } else if (roll < 10) {
+            contract.setTransportComp((45 + (roll - 6) * 5));
+        } else {
+            contract.setTransportComp(100);
+        }
+    }
+
+    public void writeToXML(final PrintWriter pw, int indent) {
+        MHQXMLUtility.writeSimpleXMLOpenTag(pw, indent++, "contractMarket");
+        MHQXMLUtility.writeSimpleXMLTag(pw, indent, "method", method.toString());
+        MHQXMLUtility.writeSimpleXMLTag(pw, indent, "lastId", lastId);
+        for (final Contract contract : contracts) {
+            contract.writeToXML(pw, indent);
+        }
+
+        for (final Integer key : clauseMods.keySet()) {
+            if (!contractIds.containsKey(key)) {
+                continue;
+            }
+
+            MHQXMLUtility.writeSimpleXMLOpenTag(pw, indent++, "clauseMods", "id", key);
+            MHQXMLUtility.writeSimpleXMLTag(pw, indent, "mods", clauseMods.get(key).mods);
+            MHQXMLUtility.writeSimpleXMLTag(pw, indent, "rerollsUsed", clauseMods.get(key).rerollsUsed);
+            MHQXMLUtility.writeSimpleXMLCloseTag(pw, --indent, "clauseMods");
+        }
+        MHQXMLUtility.writeSimpleXMLCloseTag(pw, --indent, "contractMarket");
+    }
+
+    public static AbstractContractMarket generateInstanceFromXML(Node wn, Campaign c, Version version) {
```

Comment:
There are a lot of single or two letter variable names in this method, I'm assuming it was taken from another part of the codebase and then modified to meet your needs, but it makes for really hard to read code. Especially when there are a lot of different variables to track.